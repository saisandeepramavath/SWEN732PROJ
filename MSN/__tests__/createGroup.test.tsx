import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import Groups from '../app/creategroup';
import firestore from '@react-native-firebase/firestore';
import { useRouter } from 'expo-router';

jest.mock('@expo/vector-icons', () => ({
  Ionicons: 'Ionicons',
  Entypo: 'Entypo',
  FontAwesome: 'FontAwesome',
  MaterialIcons: 'MaterialIcons',
}));

jest.mock('@react-native-firebase/app', () => ({
  __esModule: true,
  getApp: jest.fn(() => ({})), // Mock getApp to return an empty object
}));

jest.mock('@react-native-firebase/auth', () => ({
  __esModule: true,
  getAuth: jest.fn(() => ({
    currentUser: { uid: 'test-user-id' }, // Mock currentUser with a test UID
  })),
}));

jest.mock('@react-native-firebase/firestore', () => ({
  __esModule: true,
  default: jest.fn(() => ({
    collection: jest.fn(() => ({
      add: jest.fn(),
    })),
  })),
}));

jest.mock('expo-router', () => ({
  useRouter: jest.fn(),
}));

describe('Groups', () => {
  let mockRouter: any;

  beforeEach(() => {
    mockRouter = { back: jest.fn() };
    (useRouter as jest.Mock).mockReturnValue(mockRouter);
  });

  it('should match snapshot', () => {
    const { toJSON } = render(<Groups />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should display error when group name or type is missing', async () => {
    const { getByPlaceholderText, getByText, findByTestId } = render(<Groups />);

    fireEvent.changeText(getByPlaceholderText('Group name'), 'Test Group');
    fireEvent.press(getByText('Done'));

    // Verify that the error message is displayed
    expect(await findByTestId('error-message')).toHaveTextContent(
      'Please enter group name and select a type.'
    );
  });

  it('should call Firestore on successful group creation', async () => {
    const mockAdd = jest.fn();
    (firestore().collection as jest.Mock).mockReturnValue({ add: mockAdd });

    const { getByPlaceholderText, getByText, getByTestId } = render(<Groups />);

    // Simulate user input for all fields
    fireEvent.changeText(getByPlaceholderText('Group name'), 'Test Group');
    fireEvent.press(getByTestId('type-button-Trip')); // Select group type
    fireEvent.press(getByText('Done')); // Trigger group creation

  });

  it('should navigate back when Cancel is pressed', () => {
    const { getByText } = render(<Groups />);

    fireEvent.press(getByText('Cancel'));

    expect(mockRouter.back).toHaveBeenCalled();
  });
});
