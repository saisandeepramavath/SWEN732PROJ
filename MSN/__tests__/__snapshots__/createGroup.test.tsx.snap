// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Groups should match snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "backgroundColor": "#fff",
      "flexGrow": 1,
      "padding": 20,
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 30,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "color": "#009688",
              "fontSize": 16,
              "fontWeight": "500",
            }
          }
        >
          Cancel
        </Text>
      </View>
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Create a group
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "color": "#009688",
              "fontSize": 16,
              "fontWeight": "500",
            }
          }
        >
          Done
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "paddingRight": 100,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#f0f0f0",
            "borderRadius": 10,
            "display": "flex",
            "height": 70,
            "justifyContent": "center",
            "marginBottom": 20,
            "opacity": 1,
            "width": 70,
          }
        }
      >
        <Ionicons
          color="gray"
          name="camera-outline"
          size={28}
        />
      </View>
      <TextInput
        onChangeText={[Function]}
        placeholder="Group name"
        placeholderTextColor="#999"
        style={
          {
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "fontSize": 16,
            "marginBottom": 30,
            "marginLeft": 10,
            "paddingVertical": 10,
            "width": "100%",
          }
        }
        value=""
      />
    </View>
    <Text
      style={
        {
          "color": "#666",
          "fontSize": 14,
          "marginBottom": 10,
        }
      }
    >
      Type
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "gap": 10,
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#ddd",
            "borderRadius": 10,
            "borderWidth": 1,
            "opacity": 1,
            "paddingVertical": 12,
            "width": "22%",
          }
        }
        testID="type-button-Trip"
      >
        <Ionicons
          color="black"
          name="airplane"
          size={24}
        />
        <Text
          style={
            {
              "color": "#333",
              "fontSize": 13,
              "marginTop": 4,
            }
          }
        >
          Trip
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#ddd",
            "borderRadius": 10,
            "borderWidth": 1,
            "opacity": 1,
            "paddingVertical": 12,
            "width": "22%",
          }
        }
        testID="type-button-Home"
      >
        <Entypo
          color="black"
          name="home"
          size={24}
        />
        <Text
          style={
            {
              "color": "#333",
              "fontSize": 13,
              "marginTop": 4,
            }
          }
        >
          Home
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#ddd",
            "borderRadius": 10,
            "borderWidth": 1,
            "opacity": 1,
            "paddingVertical": 12,
            "width": "22%",
          }
        }
        testID="type-button-Other"
      >
        <MaterialIcons
          color="black"
          name="more-horiz"
          size={24}
        />
        <Text
          style={
            {
              "color": "#333",
              "fontSize": 13,
              "marginTop": 4,
            }
          }
        >
          Other
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
